package session_7.class_problems;

public class M4_LockerCode {

    static class Locker {

        private final int lockerNumber;
        private String combinationCode;

        // Constructor
        Locker(int lockerNumber, String combinationCode) {

            this.lockerNumber = lockerNumber;
            this.combinationCode = combinationCode;
        }

        // Change combination
        void changeCode(
                String currentCode,
                String newCode) {

            if (combinationCode.equals(currentCode)) {

                combinationCode = newCode;

                System.out.println(
                    "Code changed successfully"
                );

            } else {

                System.out.println(
                    "Code change rejected"
                );
            }
        }

        // Locker number can be read
        int getLockerNumber() {
            return lockerNumber;
        }
    }

    public static void main(String[] args) {

        Locker l = new Locker(
            101,
            "1234"
        );

        l.changeCode(
            "1234",
            "5678"
        );

        l.changeCode(
            "0000",
            "9999"
        );
    }
}